### PR TITLE
Issue 841 create form metadata when pulling from all source types

### LINKED ISSUE #841: No metadata is created when pulling from Collect or Form Definition
#### Software versions 
Briefcase v1.7

#### Problem description
No `metadata.json` file is created when pulling forms from sources other than Aggregate or Central. This produces errors when trying to fetch metadata belonging to the form key of forms pulled this way.

#### Steps to reproduce the problem
- Pull a form from Collect or Form Definition
- Export the form

#### Expected behavior
High-level behavior should be that forms pulled from Collect or Form Definition should work the same way as forms pulled from servers.

Low-level detail is that all pulled forms should be persisted in the `FormMetadataPort`, so that a `metadata.json` file is created in their form dirs.

#### Other information 
Error trace:

```
Exception in thread "Thread-4" org.opendatakit.briefcase.reused.BriefcaseException
	at java.base/java.util.Optional.orElseThrow(Optional.java:408)
	at org.opendatakit.briefcase.ui.export.ExportPanel.lambda$export$10(ExportPanel.java:198)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1654)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at org.opendatakit.briefcase.reused.job.JobsRunner.launchAsync(JobsRunner.java:63)
	at org.opendatakit.briefcase.reused.job.JobsRunner.launchAsync(JobsRunner.java:51)
	at org.opendatakit.briefcase.ui.export.ExportPanel.export(ExportPanel.java:229)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### PR DESCRIPTION
**Pull request by ggalmazor**
Closes #841

#### What has been done to verify that this works as intended?
Tried to reproduce the issue by pulling forms from Collect and Form Definition using the UI.

#### Why is this the best possible solution? Were any other approaches considered?
This is the cleanest way of solving the issue I came up with. An alternative approach would have been to change the underlying (legacy) code to deal with the form metadata port, but that felt too complicated and risky.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

TLDR:
- Users should expect to consistently see feedback from the CLI operation
- Users should expect to be able again to cancel the operation when launched from the UI

This PR is forced to improve the threading and error management code of the (legacy) pull from collect code to ensure that the process is atomic (metadata is created only when a form is successfully pulled).

This made it possible to improve user feedback on the CLI operation, since now we can free ourselves from being coupled to the EventBus in the CLI, and use the standard out directly without the complicated threading model and async task coordination from before.

We were also to bridge the legacy code's `TerminationFuture` object with the JobRunner's status object, so that the operation can be cancelled again.

#### Does this change require updates to documentation? If so, please file an issue at https://github.com/opendatakit/docs/issues/new and include the link below.
Nope.

### CODE REVIEW

**Comment by dcbriccetti** [comment]
Hi! I’m looking at this now.

**Comment by ggalmazor** [comment]
Thanks for your review, @dcbriccetti!

**Comment by kkrawczyk123** [comment]
**Tested with success!**
Verified on Ubuntu, MacOS, Windows.

verified cases:
- UI pull form odk dir
- UI pull form definition
- CMD pull from odk dir
- export form pulled from odk dir

checked for regression:
- UI pull from aggregate
- UI pull from central
- UI pull from ona
- CMD pull from aggregate
- CMD pull from central 
- export form pulled from aggregate
- export form pulled from central

@opendatakit-bot unlabel "needs testing"
@opendatakit-bot label "behavior verified"
